x-pack/legacy/plugins/siem/server/lib/detection_engine/alerts/get_filter.ts

```diff
@@ -0,0 +1,75 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { buildEsQuery } from '@kbn/es-query';
+import { Query } from 'src/plugins/data/common/query';
+import { AlertServices } from '../../../../../alerting/server/types';
+import { SignalAlertParams, PartialFilter } from './types';
+import { assertUnreachable } from '../../../utils/build_query';
+
+export const getQueryFilter = (
+  query: string,
+  language: string,
+  filters: PartialFilter[],
+  index: string[]
+) => {
+  const indexPattern = {
+    fields: [],
+    title: index.join(),
+  };
+  const queries: Query[] = [{ query, language }];
+  const config = {
+    allowLeadingWildcards: true,
+    queryStringOptions: { analyze_wildcard: true },
+    ignoreFilterIfFieldNotInIndex: false,
+    dateFormatTZ: 'Zulu',
+  };
+  const esQuery = buildEsQuery(
+    indexPattern,
+    queries,
+    filters.filter(f => f.meta != null && f.meta.disabled === false),
+    config
+  );
+  return esQuery;
+};
+
+export const getFilter = async (
```

Comment:
Nit to send this in an object. I get worried when there are so many arguments, they could be misordered

Reply:
Agreed, so changed this to use an object and it looks much cleaner now. Thanks for the tip 👍